lib/libc/termios/lib_tccheckfifo.c

```diff
@@ -0,0 +1,109 @@
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/ioctl.h>
+
+#include <termios.h>
+#include <errno.h>
+
+#include <tinyara/serial/tioctl.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Variables
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: tcrxavailable
+ *
+ * Descripton:
+ *   calls the ioctl ased on cmd,
+ *	TIOCS_AVAIL: check if there is data in rx fifo.
+ *	TIOCS_READY: check if tranmsit fifo is not full.
+ *	TIOCS_EMPTY: check if transmit fifo is empty.
+ *
+ * Input Parameters:
+ *   fd  - The 'fd' argument is an open file descriptor associated with a terminal.
+ *   cmd - ioctl argument.
+ *
+ * Returned Value:
+ *   Upon successful completion, 0 is returned. Otherwise, -1 is returned.
+ *
+ ****************************************************************************/
+
+int tccheckfifo(int fd, int cmd)
+{
+	int ret;
+
+	switch (cmd) {
```

Comment:
All cases just call the ioctl with the `cmd`.
If there is no other case for calling `ioctl` with another parameters,
how about just calling `ioctl(fd, cmd, 0);` instead of checking with switch-case?
Because in `ioctl`, it checks that `cmd` is valid or not.

Reply:
done